Library/Homebrew/dev-cmd/extract.rb

```diff
@@ -122,12 +122,15 @@ def run
         # e.g. Foo version 1.2.3 becomes FooAT123 and resides in Foo@1.2.3.rb.
         class_name = Formulary.class_s(name)
 
-        # Remove any existing version suffixes, as a new one will be added later
+        # The version can only contain digits with decimals in between.
+        version.to_s.sub!(/\D*(.+?)\D*$/, "\\1").gsub!(/\D+/, ".")
```

Comment:
`to_s` may return a copy so the `sub!` changes may be lost here

Reply:
```suggestion
        version_string = version.to_s
        												.sub(/\D*(.+?)\D*$/, "\\1")
        												.gsub(/\D+/, ".")
```